.github/workflows/Tests.yml

```diff
@@ -192,6 +221,9 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
+    - name: Fix braindead windows ${{ matrix.python-version }} on ${{ matrix.os }}
+      if: ${{ !cancelled() && runner.os == 'Windows' }}
+      run: pip install --upgrade pip
```

Comment:
@coderabbitai, I'm haveing a failures on windows-latest runners in CI related to an out-of-date `pip`.

the logs can be found below for one of the latest attempts:
[Recent CI Run](https://github.com/reactive-firewall/multicast/actions/runs/11603536984/job/32310724820?pr=175)

My observations:
* `pip` is working normally on **non-windows** runners of **same** matrix
  - Thus I suspect this is windows specific.
  - most runners are canceled when the windows-latest runner(s) fails (as configured to do so)
* `pip install --upgrade pip` is not important to the goal of the workflow, rather it is supposed to ensure some version of pip is available
* I am focusing my efforts on the job 'COVERAGE' first, (around line 226 in ".github/workflows/Tests.yml" config)
* also impacts the 'DOCS' job
* this issue has made it clear that the ".github/workflows/Tests.yml" GHA needs to be modularized in a more maintainable way after stabilizing CI

help me debug this failing GHA

Reply:
```suggestion
    - name: Fix braindead windows ${{ matrix.python-version }} on ${{ matrix.os }}
      if: ${{ !cancelled() && runner.os == 'Windows' }}
      run: python -m pip install --upgrade pip
```
🙈